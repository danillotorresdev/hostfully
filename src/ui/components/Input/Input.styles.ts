import { styled } from "@linaria/react";

export const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Label = styled.label`
  margin-bottom: 0.2rem;
  font-size: 0.8rem;
`;

export const InputField = styled.input`
  padding: 0 0.5rem;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 0.5rem;
  height: 55px;
  &:hover {
    border: 1px solid #2d2aa5;
  }
`;
