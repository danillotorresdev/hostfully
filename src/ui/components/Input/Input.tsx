import { InputHTMLAttributes, forwardRef } from "react";
import * as S from "./Input.styles";

export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
  label: string;
};

export const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ label, ...props }: InputProps, ref) => (
    <S.InputContainer>
      <S.Label htmlFor={label}>{label}</S.Label>
      <S.InputField ref={ref} {...props} id={label} />
    </S.InputContainer>
  ),
);
