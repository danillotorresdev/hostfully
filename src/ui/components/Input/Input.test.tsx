import React from "react";
import { Input } from "./Input";
import { describe, expect, it } from "vitest";
import { fireEvent, render, screen } from "@/utils/settings/tests/utilities";

describe("Input component", () => {
  it("renders input field with label correctly", () => {
    const { asFragment } = render(<Input label="Username" />);

    expect(asFragment()).toMatchSnapshot();
    
    const inputField = screen.getByLabelText("Username");
    expect(inputField).toBeDefined();

    const label = screen.getByText("Username");
    expect(label).toBeDefined();
  });

  it("forwards ref to input field", () => {
    const ref = React.createRef<HTMLInputElement>();
    render(<Input label="Username" ref={ref} />);

    expect(ref.current).toBeDefined();
  });

  it("handles user input correctly", () => {
    render(<Input label="Username" />);

    const inputField = screen.getByLabelText("Username");
    fireEvent.change(inputField, { target: { value: "testuser" } });

    expect(screen.getByDisplayValue("testuser")).toBeDefined();
  });
});
