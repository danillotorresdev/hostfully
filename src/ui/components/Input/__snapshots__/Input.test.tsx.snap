// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > renders input field with label correctly 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 i19zws4p"
  >
    <label
      class="mocked-styled-1 l5v2eyb"
      for="Username"
    >
      Username
    </label>
    <input
      class="mocked-styled-2 i1hv9n31"
      id="Username"
    />
  </div>
</DocumentFragment>
`;
